### ISSUE TITLE
Nil Error results in empty response

### ISSUE THREAD

**Issue by chowey** [chowey/jsonrpc#1]
If a nil `jsonrpc.Error` is returned from a registered method, then the result is treated as having an error and no result is returned.

Here is an example that will reproduce it:

```go
h.RegisterMethod("hello", func () (string, error) {
    var err *jsonrpc.Error
    return "Hello world!", err
})
```

Request:

```json
=> {"jsonrpc":"2.0", "method":"hello"}
```

Expected:

```json
<= {"jsonrpc":"2.0", "result":"Hello world!"}
```

Got:

```json
<= {"jsonrpc":"2.0"}
```

**Comment by chowey** [comment]
In general, [per the spec](https://www.jsonrpc.org/specification):

> **result**
> This member is REQUIRED on success.
> This member MUST NOT exist if there was an error invoking the method.

> **error**
> This member is REQUIRED on error.
> This member MUST NOT exist if there was no error triggered during invocation.

Therefore we should never have a scenario with no **result** and no **error**.

I feel that a typed nil, even if it implements the error interface, is not the same as a nil error. It is most idiomatic to report it as an error.

If in the process of calling the `Error()` method we trigger a panic, then so be it. This is a programmer error.

**Comment by chowey** [comment]
Fixed with 9fb91c374f45da31b617f68ce27117c3bed8f3ae.
